_posts/autonomy.md

```diff
@@ -0,0 +1,94 @@
+
+## What about the business?
+
+One thing that usually goes under the radar is the following sentence:
+
+> Components dealing with akin business requirements will live in the same service boundary
+
+That translates to "business requirements couple components." That's business coupling.
+
+If requirements couple components, continuing on the same path, the business couples services. They exist because the business needs them. If there is no need to deliver goods to customers, there won't be a shipping service.
+
+That being said. The next question could be:
+
+> Do they know each other?
+
+Sure they do. Otherwise, the business doesn't exist.
+
+Continuing the e-commerce analogy, if Shipping doesn't know about Sales, there wouldn't be a way to sell and deliver stuff. The business needs to sell goods, deliver them to customers, and, for example, keep under control how the overall process behaves. That is what couples together services.
+
+The mentioned business coupling in a distributed systems architecture is expressed by events crossing service boundaries. Those events are special events. [Alberto Brandolini](https://twitter.com/ziobrando), the [event storming](https://www.eventstorming.com/) inventor, calls them _Pivotal Events_. A pivotal event allows a system to move to a new state. Examples are: checkout started, order delivered, order canceled, new customer created, etc.
+
+> More details and examples about pivotal events are available in the excellent [Domain-Driven Design Use Case: Improving A Life Insurance Selling Platform](https://www.blog.jamesmichaelhickey.com/DDD-Use-Case-Life-Insurance-Platform/) by [James Hickey](https://www.blog.jamesmichaelhickey.com/).
+
+## The business doesn't care about coupling
+
+At this point, we have a problem. The business doesn't pay any attention to the type of coupling we consider harmful from the architectural perspective. Some forms of coupling might be interesting from the business perspective. For example, depending on a single supplier might be an issue. However, they don't consider the dependencies between Sales and Shipping as a concern.
+
+But we do! And that implies that we look at autonomy with different eyes.
+
+Back to the type of coupling and autonomy concerning the architecture: If events are what connect services, it's crucial to keep them as thin as possible and to respect their immutable nature. That translates into a "share nothing" policy. I expect an OrderDelivered event to look like the following:
+
+{
+  orderId;
+  customerId;
+  deliveryTypeId;
+}
+
+Usually, nothing else. That event type reduces coupling at its minimum. It's implausible that any of the shared information changes over time. The critical part is that the primary key structure is not supposed to change so we can share identifiers. Or, if that changes, it will be such a disruptive process that coupling is a breeze.
```

Comment:
```suggestion
Usually, nothing else. That event type reduces coupling to its minimum. It's improbable that any of the shared information changes over time. The critical part is that the primary key structure is not supposed to change so we can share identifiers. Or, if that changes, it will be such a disruptive process that coupling is a breeze.
```